File: src/types/reminder.ts
export interface Reminder {
  id: number;
  title: string;
  description?: string;
  trigger_at: string;
  is_recurring: boolean;
  recurrence?: string;
  is_completed: boolean;
  is_dismissed: boolean;
  snooze_until?: string;
  created_at: string;
}

export interface CreateReminder {
  title: string;
  description?: string;
  trigger_at: string;
  is_recurring?: boolean;
  recurrence?: string;
}

File: src-tauri/src/models/reminder.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Reminder {
    pub id: i64,
    pub title: String,
    pub description: Option<String>,
    pub trigger_at: String,
    pub is_recurring: bool,
    pub recurrence: Option<String>,
    pub is_completed: bool,
    pub is_dismissed: bool,
    pub snooze_until: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateReminder {
    pub title: String,
    pub description: Option<String>,
    pub trigger_at: String,
    pub is_recurring: bool,
    pub recurrence: Option<String>,
}
